overwatch/src/services/resources.rs

```diff
@@ -173,6 +173,31 @@ where
             State::from_settings(&settings)
         }
     }
+
+    /// Create a new [`ServiceResourcesHandle`](ServiceResourcesHandle) from the
+    /// current `ServiceResources`.
+    ///
+    /// It requires `inbound_relay` to be set.
+    ///
+    /// # Errors
+    ///
+    /// If the [`InboundRelay`] is not set in the `ServiceResources`.
+    pub fn as_handle(
+        &mut self,
+    ) -> Result<ServiceResourcesHandle<Message, Settings, State, RuntimeServiceId>, String> {
```

Comment:
The error here would be better under a `DynError` I think. Reason is that if at some point we migrate to a different type at least is not breaking the caller.